cord/http/utils.py

```diff
@@ -1,90 +1,107 @@
+    elif orm_class == Video:
+        is_video = True
+    else:
+        raise RuntimeError(f"Currently only `Image` or `Video` orm_class supported. Got type `{orm_class}`")
+
+    assert len(file_paths) == len(signed_urls), \
+        'Error getting the correct number of signed urls'
 
-def upload_to_signed_url(file_path, signed_url, querier, orm_class):
+    if max_workers is None:
+        max_workers = min(multiprocessing.cpu_count(), len(file_paths))
+    elif max_workers < 1:
+        raise ValueError(f"max_workers must be a positive integer. Received: {max_workers}")
+
+    orm_class_list = []
+    with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
+        total = (len(file_paths) * PROGRESS_BAR_FILE_FACTOR)
+        with tqdm(total=total, desc="Files upload progress: ") as pbar:
+            futures = []
+            for i in range(len(file_paths)):
+                file_path = file_paths[i]
+                file_name = os.path.basename(file_path)
+                signed_url = signed_urls[i]
+                assert signed_url.get('title', '') == file_name, 'Ordering issue'
+
+                future = executor.submit(_upload_single_file, file_path, signed_url, querier, orm_class, pbar,
+                                         is_video)
+                futures.append(future)
+
+            for future in concurrent.futures.as_completed(futures):
+                res = future.result()
+                orm_class_list.append(res)
+
+    return orm_class_list
+
+
+def _upload_single_file(file_path: str, signed_url: dict, querier: Querier, orm_class: OrmT, pbar,
+                        is_video: bool) -> OrmT:
+    content_type = "application/octet-stream" if is_video else mimetypes.guess_type(file_path)[0]
     res_upload = requests.put(
         signed_url.get('signed_url'),
-        data=read_in_chunks(file_path),
-        headers={'Content-Type': 'application/octet-stream'}
+        data=read_in_chunks(file_path, pbar),
+        headers={'Content-Type': content_type}
     )
-    res = None
+
     if res_upload.status_code == 200:
         data_hash = signed_url.get('data_hash')
 
         res = querier.basic_put(
             orm_class,
             uid=data_hash,
-            payload=signed_url
+            payload=signed_url,
+            enable_logging=False
         )
 
-        if orm_class(res):
-            logger.info("Successfully uploaded: %s", signed_url.get('title', ''))
-        else:
-            logger.info("Error uploading: %s", signed_url.get('title', ''))
+        if not orm_class(res):
+            logger.info("Error uploading: %s",
+                         signed_url.get('title', ''))
 
     else:
-        logger.info("Error generating signed url for: %s", signed_url.get('title', ''))
+        logger.info("Error uploading file '%s' to signed url: %s",
+                     signed_url.get('title', ''), signed_url.get('signed_url'))
+        raise RuntimeError("Bad request.")
```

Comment:
You can probably reuse the logged message in the exception.

Reply:
Done! I was sometimes discouraged to for example create an f-string which is passed into the logger, simply because if you disable `info` logging here, then passing these arguments would be a no-op (except for the `signed_url.get` I guess). Whereas if you pass an f-string, it will always be constructed, even if you're skipping that log level. I guess here it doesn't matter because I've changed the log level to `error` and will use the same string anyways.